ui/scripts/instanceWizard.js

```diff
@@ -294,53 +295,67 @@
 
             // Step 3: Service offering
             function(args) {
-                selectedTemplateObj = null; //reset
-                if (args.currentData["select-template"] == "select-template") {
-                    if (featuredTemplateObjs != null && featuredTemplateObjs.length > 0) {
-                        for (var i = 0; i < featuredTemplateObjs.length; i++) {
-                            if (featuredTemplateObjs[i].id == args.currentData.templateid) {
-                                selectedTemplateObj = featuredTemplateObjs[i];
-                                break;
+                snapshotObjs = null;
+                selectedSnapshotObj = null;
+
+                if (args.moreArgs && args.moreArgs.snapshot)
+                {
+                    zoneObjs = args.moreArgs.zone;
+                    selectedZoneObj = zoneObjs[0];
+                    hypervisorObjs = args.moreArgs.hypervisor;
+                    selectedHypervisor = hypervisorObjs[0].name;
+                    snapshotObjs = args.moreArgs.snapshot;
+                    selectedSnapshotObj = snapshotObjs[0];
+                }
+                else {
+                    selectedTemplateObj = null; //reset
+                    if (args.currentData["select-template"] == "select-template") {
+                        if (featuredTemplateObjs != null && featuredTemplateObjs.length > 0) {
+                            for (var i = 0; i < featuredTemplateObjs.length; i++) {
+                                if (featuredTemplateObjs[i].id == args.currentData.templateid) {
+                                    selectedTemplateObj = featuredTemplateObjs[i];
+                                    break;
+                                }
                             }
                         }
-                    }
-                    if (selectedTemplateObj == null) {
-                        if (communityTemplateObjs != null && communityTemplateObjs.length > 0) {
-                            for (var i = 0; i < communityTemplateObjs.length; i++) {
-                                if (communityTemplateObjs[i].id == args.currentData.templateid) {
-                                    selectedTemplateObj = communityTemplateObjs[i];
-                                    break;
+                        if (selectedTemplateObj == null) {
```

Comment:
Hi @nitin-maharana 
Couldn't this _if_ and the following _ifs_ be merged into one? 
Like:

> if ( selectedTemplateObj == null && communityTemplateObjs != null && communityTemplateObjs.length > 0)